driver/src/main/java/org/neo4j/driver/internal/net/pooling/PooledSocketConnection.java

```diff
@@ -281,6 +284,12 @@ public void setResourcesHandler( SessionResourcesHandler resourcesHandler )
     }
 
     @Override
+    public long creationTimestamp()
```

Comment:
I think addition of `PooledConnection#isValid()` will make connection too smart. It'll basically have to be aware of pool settings, do ping and stuff. I guess we could move this into existing `ConnectionValidator` interface. Maybe as a separate PR?